README.md

```diff
@@ -1,73 +1,102 @@
-# Role Name
+# crypto_policies
 
-A template for an ansible role which configures some GNU/Linux subsystem or
-service. A brief description of the role goes here.
+This Ansible role manages system-wide crypto policies.
+
+This concept is well adopted since Red Hat Enterprise Linux 8 and in Fedora.
 
 ## Requirements
 
-Any pre-requisites that may not be covered by Ansible itself or the role should
-be mentioned here. For instance, if the role uses the EC2 module, it may be a
-good idea to mention in this section that the `boto` package is required.
+The system-wide crypto policies are implemented and tested on RHEL 8/CentOS 8
+and Fedora.
+
+The role might work elsewhere, but will require adding OS definitions in
+[vars/](vars/).
 
 ## Role Variables
 
-A description of all input variables (i.e. variables that are defined in
-`defaults/main.yml`) for the role should go here as these form an API of the
-role.
+By default, this role will just report system status as described in the
+following section.
 
-Variables that are not intended as input, like variables defined in
-`vars/main.yml`, variables that are read from other roles and/or the global
-scope (ie. hostvars, group vars, etc.) can be also mentioned here but keep in
-mind that as these are probably not part of the role API they may change during
-the lifetime.
+ * `crypto_policies_policy`
 
-Example of setting the variables:
+Use this variable to specify the desired crypto policy on the target system,
+which can be either the base policy or a base policy with policy submodule(s)
+as accepted by the `update-crypto-policies` tool.For example `FUTURE` or
+`DEFAULT:NO-SHA1:GOST`. The specified base policy and policy submodule(s)
+must be available on the target system.
```

Comment:
Add "The list of available base policies can be found in the `crypto_policies_available_policies` variable and the list of available policy submodules can be found in the `crypto_policies_available_modules` variable." or something similar.
Have you considered having two variables, one `crypto_policies_policy` and the other `crypto_policies_modules`, for symmetry with the output variables?

Reply:
Added.

We did not consider this as these are used together in the API of the tool. Having the use different in the crypto policies role I would consider confusing.

Reply:
This is certainly a valid argument.

OTOH, introducing the system role is an opportunity to create a more abstract/better structured interface, not bound to a single tool. (I say "better structured", because using a text separator like `:` to separate list items in a string is odd in YAML, as it has a list type among its primitives, so you don't need to put lists in strings and parse them.) Imagine that another tool will be used to set the crypto policy in the future, with a different UI, and this variable will just look odd. I am undecided here, @richm what do you think?

Reply:
IOW, among the intended audience for a system role are administrators who neither know nor care about the UI for the underlying tool.

Reply:
> Imagine that another tool will be used to set the crypto policy in the future, with a different UI, and this variable will just look odd. I am undecided here, @richm what do you think?

This seems to me to be a case like the `selinux` module, where it is difficult to imagine a different provider being used in the `selinux` module.  I think for the initial implementation it is ok to use a string specific to the `update-crypto-policies` command in the UI.  If sometime in the future we want to support multiple implementations, it should be easy to parse the string into whatever arguments/values are used by the new implementation, or if some new version of `update-crypto-policies` takes very different values in very different formats e.g. in RHEL9 or RHEL10.

That being said - let's keep in mind that the current `update-crypto-policies` is not the only possible implementation, and let's keep the UI as general as possible.

Reply:
Right. The crypto policies package was put together quite recently and it is not expected to see different provider in future. I do not see any other way how the policies and subpolicies should be handled in future and therefore I would like to keep the same API in the role here, rather than reinventing something new.

The policy name (including colons) is something that is now documented all around RHEL and Fedora docs and if this role would require some artificial separation, I think it would cause more confusion than good and more issues when somebody would try to put colon separated values instead of arrays to the above arguments. I would rather keep it simple and if there will be a need for a new API, we can add alternative that will accept arrays.